src/Components/Components/src/CascadingQueryValueProvider.cs

```diff
@@ -0,0 +1,108 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.AspNetCore.Components.Rendering;
+using Microsoft.AspNetCore.Components.Routing;
+
+namespace Microsoft.AspNetCore.Components;
+
+internal sealed class CascadingQueryValueProvider : IComponent, ICascadingValueSupplier, IDisposable
```

Comment:
I would have expected this to be done directly within `CascadingModelBinder` and not as a separate component. `CascadingModelBinder` in my view, is responsible for all the binding operations related to `[SupplyParameterFromXXX]`